js_modules/dagit/src/schema.graphql

```diff
@@ -627,17 +633,12 @@ type Solid {
 type SolidDefinition {
   name: String!
   description: String
-  metadata: [SolidMetadataItemDefinition!]!
+  metadata: [MetadataItemDefinition!]!
```

Comment:
We can just change this later, but one cute idea to consider is having the Metadata be a union of different types, so you can have `KeyValueItem` as the base case and then other specific types for groups of well defined bundles of information. You could have an argument that controls whether you get rich types.

```
metadata { # no args just gets KV 
 ... on KeyValueItem {key value}
}

metadata(supports: ['spark', 'sql']) {
  ... on SparkMetadata {spark_fields_and_shit} 
  ... on SQLMetadata {query_string, query_string_cdn_url} 
  ... on KeyValueItem {key value} # for the rest
}
```

Something to consider if the flat key : value map becomes problematic for whatever reason.

Reply:
yup. I think figuring out our metadata plugin system is a whole subproject. glad you are thinking along these lines.